consensus/snapshot/snapshot_metadata.c

```diff
@@ -81,6 +82,8 @@ retcode_t iota_snapshot_metadata_serialize_str(snapshot_metadata_t const *const
     offset += value_len;
   }
 
+  memcpy(str + offset, "\0", 1);
```

Comment:
Maybe don't call `memcpy` for only one char and set it directly ? `str[offset] = '\0'`